### ISSUE TITLE
CW attack doesn't work.maybe i don't know how to run correctly. help me

### ISSUE THREAD

**Issue by xieshisheng** [Harry24k/adversarial-attacks-pytorch#5]

**Comment by Harry24k** [comment]
Please let me know the error situation.

**Comment by KurtSunxx** [comment]
Sir, I also encounter this issue, I found when I use the parameters default in your code, the success rate of  adversarial attack is close to 0. Cause I do not check the original paper about C&W, so I am not sure how to do. Can you help me ? @Harry24k

**Comment by Harry24k** [comment]
> Sir, I also encounter this issue, I found when I use the parameters default in your code, the success rate of adversarial attack is close to 0. Cause I do not check the original paper about C&W, so I am not sure how to do. Can you help me ? @Harry24k

[With default c, you can't easily get adversarial images. Set higher c like 1.](https://github.com/Harry24k/adversarial-attacks-pytorch/blob/55d7d0c269d30418150aa828d2ff029c2d09378a/torchattacks/attacks/cw.py#L27)

**Comment by KurtSunxx** [comment]
Ok,  I will try it now 1137582789 Email:1137582789@qq.com Signature is customized by Netease Mail Master On 07/30/2020 22:04, Harry Kim wrote: Sir, I also encounter this issue, I found when I use the parameters default in your code, the success rate of adversarial attack is close to 0. Cause I do not check the original paper about C&W, so I am not sure how to do. Can you help me ? @Harry24k With default c, you can't easily get adversarial images. Set higher c like 1. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by xieshisheng** [comment]
set c like 1, I tried, but the code doesn't work too.I also check the souce code, i found the source code are not wrong.So i dont know where is wrong.

**Comment by xieshisheng** [comment]
I found, set c to 10 or up(1 is not enough), CW attack can changed image to adversarial image with high  success rate

**Comment by KurtSunxx** [comment]
Let me try, thanks for your precious experience! @xieshisheng

**Comment by KurtSunxx** [comment]
By the way, sir, how do you set iters and lr? @xieshisheng

**Comment by xieshisheng** [comment]
I dont change iters and lr.
source code 
"loss1 = nn.MSELoss(reduction='sum')(a, images)
 loss2 = torch.sum(self.c*f(a))"
 i think MSEELoss force new image as same as old image, if c is small ,gradient of loss2 is small . So the "adversarial" image is just as same as image, but not truly adversarial image.

**Comment by KurtSunxx** [comment]
I cannot agree with you any more. But I think the iteration maybe to large...... 1137582789 Email:1137582789@qq.com Signature is customized by Netease Mail Master On 07/31/2020 10:05, xieshisheng wrote: I dont change iters and lr. source code "loss1 = nn.MSELoss(reduction='sum')(a, images) loss2 = torch.sum(self.c*f(a))" i think MSEELoss force new image as same as old image, if c is small ,gradient of loss2 is small . So the "adversarial" image is just as same as image, but not truly adversarial image. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by xieshisheng** [comment]
hhhhh, maybe large iteration is more  effective.I‘ll try it later.thank you

**Comment by KurtSunxx** [comment]
thanks too! 1137582789 Email:1137582789@qq.com Signature is customized by Netease Mail Master On 07/31/2020 10:31, xieshisheng wrote: hhhhh, maybe large iteration is more effective.I‘ll try it later.thank you — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.
